app/controllers/engagements_controller.rb

```diff
@@ -45,6 +48,28 @@ def destroy
     redirect_to @app, notice: 'Engagement was successfully destroyed.'
   end
 
+  def export
+    # engagement.iterations returns list of iterations
+    # create a hash of iterations
+    iterations_hash = Hash.new
+    @engagement.iterations.each do |i|
+      iterations_hash[i.id] = i.as_json
+    end
+
+    # turn engagement into a hash first before adding more stuff to it
+    eng_hash = @engagement.as_json
+    eng_hash["iterations"] = iterations_hash
+    # make entire thing into a json
+    eng_params = JSON.parse(eng_hash.to_json)
+    keys = eng_params.keys
+    values = eng_params.values
+    file = CSV.generate do |csv|
+      csv << keys
+      csv << values
+    end
+    send_data(file, :filename => "engagement.csv")
```

Comment:
Might be a good idea to have different file names for each engagement. e.g. file name by engagement id. But push to next iteration!

Reply:
yes that's a good idea!